src/clzones.cpp

```diff
@@ -11,61 +11,59 @@
 #include <iostream>
 #include <fstream>
 
-clZones::clZones()
+zone_manager::zone_manager()
 {
-    //Add new zone types here
-    //Use: if (g->checkZone("ZONE_NAME", posx, posy)) {
-    //to check for a zone
-
-    vZoneTypes.push_back(std::make_pair(_("No Auto Pickup"), "NO_AUTO_PICKUP"));
+    types.push_back( std::make_pair( _("No Auto Pickup"), "NO_AUTO_PICKUP" ) );
+    types.push_back( std::make_pair( _("No NPC Pickup"),  "NO_NPC_PICKUP" ) );
 }
 
-void clZones::clZoneData::setName()
+void zone_manager::zone_data::set_name()
 {
-    const std::string sName = string_input_popup(_("Zone name:"), 55, this->sName, "", "", 15);
+    const std::string new_name = string_input_popup( _( "Zone name:" ), 55, name, "", "", 15 );
 
-    this->sName = (sName == "") ? _("<no name>") : sName;
+    name = ( new_name == "" ) ? _( "<no name>" ) : new_name;
 }
 
-void clZones::clZoneData::setZoneType(std::vector<std::pair<std::string, std::string> >
-                                      vZoneTypes)
+void zone_manager::zone_data::set_zone_type()
 {
+    const auto &types = get_manager().get_zone_types();
     uimenu as_m;
-    as_m.text = _("Select zone type:");
+    as_m.text = _( "Select zone type:" );
 
-    for (unsigned int i = 0; i < vZoneTypes.size(); ++i) {
-        as_m.entries.push_back(uimenu_entry(i + 1, true, (char)i + 1, vZoneTypes[i].first));
+    for( size_t i = 0; i < types.size(); ++i ) {
+        as_m.entries.push_back( uimenu_entry( i + 1, true, ( char )i + 1, types[i].first ) );
```

Comment:
Why not use `uimenu::addentry`? Also `( char )i + 1` seems *very* wrong. It starts with `(char)1`, which is a control character and not printable. Why not use `MENU_AUTOASSIGN`? And finally: why the `+1` at all? That offset is not needed. It can be removed here and below when referring to `as_m.ret`.